sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamAggregationRel.java

```diff
@@ -84,6 +85,31 @@ public BeamAggregationRel(
     this.windowFieldIndex = windowFieldIndex;
   }
 
+  public BeamAggregationRel(
+      RelOptCluster cluster,
+      RelTraitSet traits,
+      RelNode child,
+      RelDataType rowType,
+      boolean indicator,
+      ImmutableBitSet groupSet,
+      List<ImmutableBitSet> groupSets,
+      List<AggregateCall> aggCalls,
+      @Nullable WindowFn<Row, IntervalWindow> windowFn,
+      int windowFieldIndex) {
+    this(
+        cluster,
+        traits,
+        child,
+        indicator,
+        groupSet,
+        groupSets,
+        aggCalls,
+        windowFn,
+        windowFieldIndex);
+
+    this.rowType = rowType;
```

Comment:
Where is this rowType used?

Reply:
RelNode type is already inferred from input nodes. Usually when you need to use it, you can use getRowType() function to get it than save it as a class member.

Reply:
There is a JIRA issue: https://jira.apache.org/jira/browse/BEAM-7609, when running queries with "SELECT DISTINCT + JOIN", resulting field names are not assigned proper name.
Even though it does not solve this particular issue, preserving the rowType should not hurt (where previously it would just ignore it and set it to null).

Reply:
I see, in that case I will remove this constructor. Do you think adding `deriveRowType();` to the original constructor makes sense or it would be redundant?

Reply:
I think `deriveRowType()` can be called when it's needed?

Reply:
Agreed, reverted BeamAggregationRel back.